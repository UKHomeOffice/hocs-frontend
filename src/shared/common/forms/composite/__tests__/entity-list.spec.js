import React from 'react';
import EntityList from '../entity-list.jsx';

const FIELD_NAME = 'entity-list';

const choices = [
    { label: 'isA', value: 'A' },
    { label: 'isB', value: 'B' },
    { label: 'isC', value: 'C' }
];

describe('Document add component', () => {

    it('should render with default props', () => {
        const wrapper = render(<EntityList name={FIELD_NAME} updateState={() => null}/>);
        expect(wrapper).toBeDefined();
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with value when passed', () => {
        expect(
            render(<EntityList name={FIELD_NAME} choices={choices} updateState={() => null} value="A" />)
        ).toMatchSnapshot();
    });

    it('should render default with first option when no value passed', () => {
        expect(
            render(<EntityList name={FIELD_NAME} choices={choices} updateState={() => null} />)
        ).toMatchSnapshot();
    });

    it('should render disabled when passed', () => {
        const wrapper = mount(<EntityList name={FIELD_NAME} updateState={() => null} disabled={true}/>);
        expect(wrapper).toBeDefined();
        expect(wrapper.find('EntityList').props().disabled).toEqual(true);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with correct label when passed', () => {
        const label = 'MY_LABEL'
        const wrapper = mount(<EntityList name={FIELD_NAME} updateState={() => null} label={label}/>);
        expect(wrapper).toBeDefined();
        expect(wrapper.find('EntityList').props().label).toEqual(label);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with validation error when passed', () => {
        const error = 'MY_ERROR';
        const wrapper = mount(<EntityList name={FIELD_NAME} updateState={() => null} error={error}/>);
        expect(wrapper).toBeDefined();
        expect(wrapper.find('EntityList').props().error).toEqual(error);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with hint when passed', () => {
        const hint = 'MY_HINT';
        const wrapper = mount(<EntityList name={FIELD_NAME} choices={choices} updateState={() => null} hint={hint}/>);
        expect(wrapper).toBeDefined();
        expect(wrapper.find('EntityList').props().hint).toEqual(hint);
        expect(wrapper).toMatchSnapshot();

    });

    it('should execute callback on initialization', () => {
        const mockCallback = jest.fn();
        shallow(
            <EntityList name={FIELD_NAME} updateState={mockCallback} />
        );
        expect(mockCallback).toHaveBeenCalledTimes(1);
        expect(mockCallback).toHaveBeenCalledWith({ [FIELD_NAME]: null });
    });

    it('should execute callback on change', () => {
        const mockCallback = jest.fn();
        const wrapper = shallow(
            <EntityList name={FIELD_NAME} choices={choices} updateState={mockCallback} />
        );
        mockCallback.mockReset();

        const firstValue = 'A';
        wrapper.find(`#${FIELD_NAME}-${firstValue}`).simulate('change', { target: { value: firstValue } });
        expect(mockCallback).toHaveBeenCalledTimes(1);
        expect(mockCallback).toHaveBeenCalledWith({ [FIELD_NAME]: firstValue });

        const secondValue = 'B';
        wrapper.find(`#${FIELD_NAME}-${secondValue}`).simulate('change', { target: { value: secondValue } });
        expect(mockCallback).toHaveBeenCalledTimes(2);
        expect(mockCallback).toHaveBeenCalledWith({ [FIELD_NAME]: secondValue });
    });

});