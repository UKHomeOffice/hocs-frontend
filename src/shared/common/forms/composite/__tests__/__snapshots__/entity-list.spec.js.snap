// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document add component should render default with first option when no value passed 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
  >
    <legend
      class="govuk-fieldset__legend"
      id="entity-list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      >
        Primary undefined
      </span>
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-radios govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <div
              class="govuk-radios__item"
            >
              <input
                checked=""
                class="govuk-radios__input"
                id="entity-list-A"
                name="entity-list"
                type="radio"
                value="A"
              />
              <label
                class="govuk-label govuk-radios__label"
                for="entity-list-A"
              >
                isA
              </label>
            </div>
          </td>
          <td
            class="govuk-table__cell"
          />
        </tr>
        <tr
          class="govuk-radios govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <div
              class="govuk-radios__item"
            >
              <input
                class="govuk-radios__input"
                id="entity-list-B"
                name="entity-list"
                type="radio"
                value="B"
              />
              <label
                class="govuk-label govuk-radios__label"
                for="entity-list-B"
              >
                isB
              </label>
            </div>
          </td>
          <td
            class="govuk-table__cell"
          />
        </tr>
        <tr
          class="govuk-radios govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <div
              class="govuk-radios__item"
            >
              <input
                class="govuk-radios__input"
                id="entity-list-C"
                name="entity-list"
                type="radio"
                value="C"
              />
              <label
                class="govuk-label govuk-radios__label"
                for="entity-list-C"
              >
                isC
              </label>
            </div>
          </td>
          <td
            class="govuk-table__cell"
          />
        </tr>
      </tbody>
    </table>
  </fieldset>
</div>
`;

exports[`Document add component should render disabled when passed 1`] = `
<EntityList
  choices={Array []}
  disabled={true}
  name="entity-list"
  type="radio"
  updateState={[Function]}
>
  <div
    className="govuk-form-group"
  >
    <fieldset
      className="govuk-fieldset "
      disabled={true}
    >
      <legend
        className="govuk-fieldset__legend"
        id="entity-list-legend"
      >
        <span
          className="govuk-fieldset__heading govuk-label--s"
        >
          Primary undefined
        </span>
      </legend>
      <table
        className="govuk-table"
      >
        <tbody
          className="govuk-table__body"
        >
          No EntityList
        </tbody>
      </table>
    </fieldset>
  </div>
</EntityList>
`;

exports[`Document add component should render with correct label when passed 1`] = `
<EntityList
  choices={Array []}
  disabled={false}
  label="MY_LABEL"
  name="entity-list"
  type="radio"
  updateState={[Function]}
>
  <div
    className="govuk-form-group"
  >
    <fieldset
      className="govuk-fieldset "
      disabled={false}
    >
      <legend
        className="govuk-fieldset__legend"
        id="entity-list-legend"
      >
        <span
          className="govuk-fieldset__heading govuk-label--s"
        >
          Primary MY_LABEL
        </span>
      </legend>
      <table
        className="govuk-table"
      >
        <tbody
          className="govuk-table__body"
        >
          No EntityList
        </tbody>
      </table>
    </fieldset>
  </div>
</EntityList>
`;

exports[`Document add component should render with default props 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
  >
    <legend
      class="govuk-fieldset__legend"
      id="entity-list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      >
        Primary undefined
      </span>
    </legend>
    No EntityList
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      />
    </table>
  </fieldset>
</div>
`;

exports[`Document add component should render with hint when passed 1`] = `
<EntityList
  choices={
    Array [
      Object {
        "label": "isA",
        "value": "A",
      },
      Object {
        "label": "isB",
        "value": "B",
      },
      Object {
        "label": "isC",
        "value": "C",
      },
    ]
  }
  disabled={false}
  hint="MY_HINT"
  name="entity-list"
  type="radio"
  updateState={[Function]}
>
  <div
    className="govuk-form-group"
  >
    <fieldset
      className="govuk-fieldset "
      disabled={false}
    >
      <legend
        className="govuk-fieldset__legend"
        id="entity-list-legend"
      >
        <span
          className="govuk-fieldset__heading govuk-label--s"
        >
          Primary undefined
        </span>
      </legend>
      <span
        className="govuk-form-hint"
      >
        MY_HINT
      </span>
      <table
        className="govuk-table"
      >
        <tbody
          className="govuk-table__body"
        >
          <tr
            className="govuk-radios govuk-table__row"
            key="0"
          >
            <td
              className="govuk-table__cell"
            >
              <div
                className="govuk-radios__item"
              >
                <input
                  checked={true}
                  className="govuk-radios__input"
                  id="entity-list-A"
                  name="entity-list"
                  onChange={[Function]}
                  type="radio"
                  value="A"
                />
                <label
                  className="govuk-label govuk-radios__label"
                  htmlFor="entity-list-A"
                >
                  isA
                </label>
              </div>
            </td>
            <td
              className="govuk-table__cell"
            />
          </tr>
          <tr
            className="govuk-radios govuk-table__row"
            key="1"
          >
            <td
              className="govuk-table__cell"
            >
              <div
                className="govuk-radios__item"
              >
                <input
                  checked={false}
                  className="govuk-radios__input"
                  id="entity-list-B"
                  name="entity-list"
                  onChange={[Function]}
                  type="radio"
                  value="B"
                />
                <label
                  className="govuk-label govuk-radios__label"
                  htmlFor="entity-list-B"
                >
                  isB
                </label>
              </div>
            </td>
            <td
              className="govuk-table__cell"
            />
          </tr>
          <tr
            className="govuk-radios govuk-table__row"
            key="2"
          >
            <td
              className="govuk-table__cell"
            >
              <div
                className="govuk-radios__item"
              >
                <input
                  checked={false}
                  className="govuk-radios__input"
                  id="entity-list-C"
                  name="entity-list"
                  onChange={[Function]}
                  type="radio"
                  value="C"
                />
                <label
                  className="govuk-label govuk-radios__label"
                  htmlFor="entity-list-C"
                >
                  isC
                </label>
              </div>
            </td>
            <td
              className="govuk-table__cell"
            />
          </tr>
        </tbody>
      </table>
    </fieldset>
  </div>
</EntityList>
`;

exports[`Document add component should render with validation error when passed 1`] = `
<EntityList
  choices={Array []}
  disabled={false}
  error="MY_ERROR"
  name="entity-list"
  type="radio"
  updateState={[Function]}
>
  <div
    className="govuk-form-group govuk-form-group--error"
  >
    <fieldset
      className="govuk-fieldset "
      disabled={false}
    >
      <legend
        className="govuk-fieldset__legend"
        id="entity-list-legend"
      >
        <span
          className="govuk-fieldset__heading govuk-label--s"
        >
          Primary undefined
        </span>
      </legend>
      <span
        className="govuk-error-message"
      >
        MY_ERROR
      </span>
      <table
        className="govuk-table"
      >
        <tbody
          className="govuk-table__body"
        >
          No EntityList
        </tbody>
      </table>
    </fieldset>
  </div>
</EntityList>
`;

exports[`Document add component should render with value when passed 1`] = `
<div
  class="govuk-form-group"
>
  <fieldset
    class="govuk-fieldset "
  >
    <legend
      class="govuk-fieldset__legend"
      id="entity-list-legend"
    >
      <span
        class="govuk-fieldset__heading govuk-label--s"
      >
        Primary undefined
      </span>
    </legend>
    <table
      class="govuk-table"
    >
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-radios govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <div
              class="govuk-radios__item"
            >
              <input
                checked=""
                class="govuk-radios__input"
                id="entity-list-A"
                name="entity-list"
                type="radio"
                value="A"
              />
              <label
                class="govuk-label govuk-radios__label"
                for="entity-list-A"
              >
                isA
              </label>
            </div>
          </td>
          <td
            class="govuk-table__cell"
          />
        </tr>
        <tr
          class="govuk-radios govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <div
              class="govuk-radios__item"
            >
              <input
                class="govuk-radios__input"
                id="entity-list-B"
                name="entity-list"
                type="radio"
                value="B"
              />
              <label
                class="govuk-label govuk-radios__label"
                for="entity-list-B"
              >
                isB
              </label>
            </div>
          </td>
          <td
            class="govuk-table__cell"
          />
        </tr>
        <tr
          class="govuk-radios govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            <div
              class="govuk-radios__item"
            >
              <input
                class="govuk-radios__input"
                id="entity-list-C"
                name="entity-list"
                type="radio"
                value="C"
              />
              <label
                class="govuk-label govuk-radios__label"
                for="entity-list-C"
              >
                isC
              </label>
            </div>
          </td>
          <td
            class="govuk-table__cell"
          />
        </tr>
      </tbody>
    </table>
  </fieldset>
</div>
`;
